x/concentrated-liquidity/math/math_test.go

```diff
@@ -277,6 +319,27 @@ func TestCalcAmount1Delta(t *testing.T) {
 			roundUp:         true,
 			amount1Expected: osmomath.MustNewBigDecFromStr("28742157707995443393876876754535992.801567623738751734").Ceil(), // round up at precision end.
 		},
+		"low price range (no round up)": {
+			liquidity: smallLiquidity,
+			sqrtPA:    sqrtANearMin,
+			sqrtPB:    sqrtBNearMin,
+			roundUp:   false,
+			// from clmath decimal import *
+			// from math import *
+			// calc_amount_one_delta(liq, sqrtPriceA, sqrtPriceB, False)
+			amount1Expected: osmomath.MustNewBigDecFromStr("0.000000000000000000000000000103787162"),
+		},
+		"low price range (with round up)": {
+			liquidity: smallLiquidity,
+			sqrtPA:    sqrtANearMin,
+			sqrtPB:    sqrtBNearMin,
+			roundUp:   true,
+			// from clmath decimal import *
+			// calc_amount_one_delta(liq, sqrtPriceA, sqrtPriceB, False)
+			// Actual result: 0.000000000000000000000000000103787163
+			// Gets rounded up to 1. Is this acceptable when the multiplicative difference is so large?
+			amount1Expected: osmomath.MustNewBigDecFromStr("0.000000000000000000000000000103787163").Ceil(),
+		},
```

Comment:
My recollection of digging into this during prelaunch internal reviews was that this was at least not a safety issue. We probably want to minimize rounding error in general but it's probably okay to overround to be safe.

Reply:
> This seems problematic right? Shouldn't any calculation for amount1 being less than 1 just fail?

No, there are many instances where it is valid. E.g. no liquidity, adjacent ticks with little liquidity.

On another look, I also think this is fine since done only in pool's favor (controlled by the `roundUp` flag)

I also reviewed Uniswap's logic and their rounding is even stricter where they take a ceiling of an intermediary calculation